pkg/expression/scalar_function.go

```diff
@@ -741,6 +741,17 @@ func ReHashCode(sf *ScalarFunction) {
 		evalTp := sf.RetType.EvalType()
 		sf.hashcode = append(sf.hashcode, byte(evalTp))
 	}
+	if sf.FuncName.L == ast.Grouping {
+		sf.hashcode = codec.EncodeInt(sf.hashcode, int64(sf.Function.(*BuiltinGroupingImplSig).GetGroupingMode()))
+		marks := sf.Function.(*BuiltinGroupingImplSig).GetMetaGroupingMarks()
+		sf.hashcode = codec.EncodeInt(sf.hashcode, int64(len(marks)))
+		for _, mark := range marks {
+			sf.hashcode = codec.EncodeInt(sf.hashcode, int64(len(mark)))
+			for k := range mark {
+				sf.hashcode = codec.EncodeInt(sf.hashcode, int64(k))
+			}
```

Comment:
In golang, the order of iterating a map with `for ... range` is nondeterministic, so one expression will produce different results.
I think we need to sort it before encoding.